File: 031_Konstruktori/src/hr/unipu/java/topic3/constructors/Main.java
package hr.unipu.java.topic3.constructors;

public class Main {

    public static void main(String[] args) {

        Point p = new Point();          // Instantation with No-argument constructor

        Point p1 = new Point(2, 5);    // Instantation with constructor
        Point p2 = new Point(p1);            // Copy constructor (deep copy)

        //Point p2;
        //p2 = p1;

        Math.cos(Math.PI);

        p1.setX(1);
        p1.setY(2);
        System.out.println(p1);
        System.out.println(p2);
    }
}

File: 031_Konstruktori/src/hr/unipu/kotlin/topic3/constructors/Main.kt
package hr.unipu.kotlin.topic3.constructors

fun main(args: Array<String>) {

    val p = Point()          // Instantation with No-argument constructor

    val p1 = Point(2.0, 5.0)    // Instantation with constructor
    val p2 = Point(p1)            // Copy constructor (deep copy)

    //Point p2;
    //p2 = p1;

    Math.cos(Math.PI)

    p1.x = 1.0
    p1.y = 2.0

    println(p1)
    println(p2)
    // >   (1.0, 2.0)
    // >   (2.0, 5.0)
}